### ISSUE TITLE
⚠️Run fails after v3.1.0 in `editorconfig-checker.javascript`

### ISSUE THREAD

**Issue by lumirlumir** [editorconfig-checker/editorconfig-checker#409]
Hello,

I noticed that version 3.1.0 doesn’t include binary files, which causes CI to fail when running `editorconfig-checker` in a JavaScript environment.  

It seems that `editorconfig-checker.javascript` depends on the GitHub Release Assets, so the binaries should be included.

![image](https://github.com/user-attachments/assets/c49f7969-17bf-4e1b-9c6c-fa010df4b56a)

- Code that I referenced

    - https://github.com/editorconfig-checker/editorconfig-checker.javascript/blob/master/src/index.ts

    - https://github.com/editorconfig-checker/editorconfig-checker.javascript/blob/master/src/constants.ts

- Error Message

    ![image](https://github.com/user-attachments/assets/a73896db-eba2-492b-adc6-43c273115129)

**Comment by A1bi** [comment]
Release pipeline is [broken](https://github.com/editorconfig-checker/editorconfig-checker/actions/runs/12666582550).

**Comment by mstruebing** [comment]
@klaernie @theoludwig do you know if the pipeline uses/have access to the correct secrets?

**Comment by lumirlumir** [comment]
If you're using `editorconfig-checker.javascript`, make sure to add `EC_VERSION` to your `process.env`. This will ensure it works correctly.

- **Example: GitHub Actions**

    ```yaml
    env:
          EC_VERSION: v3.0.3
    ```

**Comment by ngxson** [comment]
For github workflow, you can specify the version number via `inputs`:

```yaml
jobs:
  editorconfig:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: editorconfig-checker/action-editorconfig-checker@v2
        with:
          version: v3.0.3
      - run: editorconfig-checker
```

**Comment by klaernie** [comment]
> @klaernie @theoludwig do you know if the pipeline uses/have access to the correct secrets?

Access to the secrets is not the problem, there was no complaint that there were missing values. I would assume the username/password combination stored in the secrets is not working. Assuming you(@mstruebing) created them, it would be worth a try to reenter the secrets on organization level and remove the secrets on repo level.

**Comment by mstruebing** [comment]
@klaernie I've updated them in both places just to be sure.
But as far as I can see the data should not have changed.

**Comment by echoix** [comment]
There's also all the related repos that aren't updated too.
No mention of pre-commit?

**Comment by theoludwig** [comment]
> @klaernie @theoludwig do you know if the pipeline uses/have access to the correct secrets?

I don't know what happened, or why the secrets changed.
I saw the error when I merged the release PR this morning, and posted a comment here: https://github.com/editorconfig-checker/editorconfig-checker/pull/408#issuecomment-2577019649

>  I've updated them in both places just to be sure.

I re-run the goreleaser CI, it seems to have worked: https://github.com/editorconfig-checker/editorconfig-checker/actions/runs/12666582550/job/35314610695

**Comment by theoludwig** [comment]
@ngxson @lumirlumir

Thank you for providing the workarounds, and sorry for the inconveniences.
I would say, it is a best practice to keep the version explicit. :+1:

The issue has been resolved, and you can now all upgrade to v3.1.0, it should work without issues.

**Comment by theoludwig** [comment]
In the GitHub release, I see lot of `.json` files in the assets, are they necessary? Maybe we can remove them, so that there are less assets, it feels a bit much right now. What do you think? @klaernie

**Comment by mstruebing** [comment]
Maybe the bot user needs to be assigned a specific organisation group?
I’ve updated the secrets but I really doubt they have changed.

**Comment by lumirlumir** [comment]
Thank you, everyone, for your hard work!🔥🔥

**Comment by klaernie** [comment]
> In the GitHub release, I see lot of `.json` files in the assets, are they necessary? Maybe we can remove them, so that there are less assets, it feels a bit much right now. What do you think? @klaernie

@theoludwig I would expect most people do not care. Currently the list is more bloated, since we have both files with the old binary name `ec`as well as archives using the new name `editorconfig-checker`. Once #255 is merged this will be better as well.
